DephyInc/Actuator-Package
# [DSEP-128] Repo Cleanup, githooks, PR template

# Description

Adds githooks with autoformatting and auto-linting for python. Fixes problematic files.

# Changes

* [DSEP-128] Adds githooks
* Adds githooks installer
* Adds PR template
* Updates readme with instructions

# Test Plan

1. See the `README.md` file and install githooks according to the instructions
2. Run the pre-commit checks manually according to those instructions.

# Expected Results

* The pre-commit checks will run in a few seconds (<10s)
* The output will look like this:
```
Trim Trailing Whitespace.................................................Passed
Fix End of Files.........................................................Passed
Check Yaml...............................................................Passed
Check for added large files..............................................Passed
Check that executables have shebangs.....................................Passed
Mixed line ending........................................................Passed
Check for merge conflicts................................................Passed
Check docstring is first.................................................Passed
Check for case conflicts.................................................Passed
CleverCSV Standardize................................(no files to check)Skipped
Forbid spaces in indentation.............................................Passed
black-but-with-tabs......................................................Passed
pylint...................................................................Passed
```
* Once installed, the chack will run prior to making any commit and will prevent you from committing if the checks fail.

# Requirements 
Edit the list accordingly.
- [ ] Tested on RPi.
- [ ] Connected ActPack and runs scripts.


[DSEP-128]: https://dephyinc.atlassian.net/browse/DSEP-128
